ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/TermEvaluationContext.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter;
+
+import io.confluent.ksql.GenericRow;
+
+public final class TermEvaluationContext {
```

Comment:
Just wondering, do you have any plans to extend this class in the future to include other metadata? Since right now we can just remove the `TermEvaluationContext.of` call to create them -- though JVM's GC today is smart enough to not get impacted for such short-lived objects anyways, so this is really just to dig into your head for future plans :)

Reply:
I wanted to allow for the possibility that maybe there are more fields in the future, and they may not all come together.  Either way, I can get rid of the factory method, since it doesn't add anything now.